File: src/leetcode/2026/04/Solution747.cpp
#include <vector>
#include <iostream>
using namespace std;

class Solution747 {
public:
    int dominantIndex(vector<int>& nums) {
        if (nums.empty()) {
            return -1;
        }
        int maxIndex = 0;
        for (int i = 1; i < nums.size(); ++i) {
            if (nums[i] > nums[maxIndex]) {
                maxIndex = i;
            }
        }
        for (int i = 0; i < nums.size(); ++i) {
            if (i != maxIndex && nums[i] * 2 > nums[maxIndex]) {
                return -1;
            }
        }
        return maxIndex;
    }
};

int main() {
    Solution747 solution;

    vector<int> nums1 = {3, 6, 1, 0};
    cout << solution.dominantIndex(nums1) << endl; // Output: 1

    vector<int> nums2 = {1, 2, 3, 4};
    cout << solution.dominantIndex(nums2) << endl; // Output: -1

    vector<int> nums3 = {1};
    cout << solution.dominantIndex(nums3) << endl; // Output: 0

    return 0;
}

File: src/leetcode/2026/04/Solution747.java
public class Solution747 {
    public int dominantIndex(int[] nums) {
        if (nums == null || nums.length == 0) {
            return -1;
        }
        int maxIndex = 0;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] > nums[maxIndex]) {
                maxIndex = i;
            }
        }
        for (int i = 0; i < nums.length; i++) {
            if (i != maxIndex && nums[i] * 2 > nums[maxIndex]) {
                return -1;
            }
        }
        return maxIndex;
    }

    public static void main(String[] args) {
        Solution747 solution = new Solution747();

        int[] nums = { 3, 6, 1, 0 };
        int output = solution.dominantIndex(nums);
        System.out.println("Output: " + output);

        nums = new int[] { 1, 2, 3, 4 };
        output = solution.dominantIndex(nums);
        System.out.println("Output: " + output);

        nums = new int[] { 1 };
        output = solution.dominantIndex(nums);
        System.out.println("Output: " + output);
    }
}
